quartz/components/scripts/util.ts

```diff
@@ -24,3 +24,21 @@ export function removeAllChildren(node: HTMLElement) {
     node.removeChild(node.firstChild)
   }
 }
+
+// AliasRedirect emits HTML redirects which also have the link[rel="canonical"]
+// containing the URL it's redirecting to.
+// Extracting it here with regex is _probably_ faster than parsing the entire HTML
+// with a DOMParser effectively twice (here and later in the SPA code), even if
+// way less robust - we only care about our own generated redirects after all.
+const canonicalRegex = /<link rel="canonical" href="([^"]*)">/
+
+export async function fetchCanonical(url: URL): Promise<Response> {
+  return fetch(`${url}`).then(async (res) => {
+    if (!res.headers.get("content-type")?.startsWith("text/html")) {
+      return res
+    }
+    const text = await res.clone().text()
+    const [_, redirect] = text.match(canonicalRegex) ?? []
+    return redirect ? fetch(redirect) : res
+  })
```

Comment:
not entirely sure about the .then pattern but since we need to await the onFullfill then why don't just await the fetch then just do it in the async function?

Reply:
You're absolutely correct, I got to that state after some refactors or something and didn't notice that

Reply:
Done, also added a comment for why `res.clone()`